submissions/ALT+F4 LTN-RLD-Cybersyn Mod Comparison/Requesting a Train.md

```diff
@@ -0,0 +1,111 @@
+
+Much like Cybersyn, Rail Logistics Dispatcher forgoes replacing the vanilla Train Stop in favor of adding another entity to control it. However, the Dispatcher it adds is smaller, doesn't require power, and uses lamp graphics similar to LTN, making it easier to tell whether the stop is functioning correctly. Taken together, the minimum footprint for the dispatcher and train stop is 2x2 + 1x1, with the latter needing to be 1 tile away from the stop it controls.
+
+{% include image.html src='RLD-GUI.png' alt='Rail Logistics Dispather's GUI' caption='The GUI used to configure the Dispatcher' %}
+
+While add-on mods like [LTN Combinator Modernized](https://mods.factorio.com/mod/LTN_Combinator_Modernized) have been very helpful to those lacking experience with circuit network, the fact remains that it is an entirely separate mod dedicated to simplfying and improving the user experience. With that in mind, RLD opted to include a GUI right out of the box. (RLD does also offer a 'signals mode' to allow the user to configure the Dispatcher in a manner more similar to LTN if desired.)
+
+Unlike both of the previous two mods, Rail Logistics Dispatcher adds no new signals at all. Everything that was done through signals such as Thresholds, Train Configurations, Network IDs, are instead done through use of the GUI. RLD also goes all-in on the the per-item thresholds. It doesn't have any per-station thresholds through its GUI, nor any global thresholds in its settings. However, it can save certain Requests and Thresholds to be applied again. In addition to the individual item units and stacks of items, RLD also has the ability to set requests based on a wagon (fluid or cargo) load.
+
+Instead of the "Locked Slots Signal" that LTN and CYbersyn share, RLD instead opts to reverse inserters and have them put items back inside the chest or warehouse from which they picked it up. On one hand, this does mean that the user would have difficulty making stations using loaders or inserters picking up directly from belts. On the other, this does mean that multi-item provider trains can fill trains with slightly higher loads.
+
+Rail Logistics Dispatcher's GUI makes it the easiest of the three mods to use, but it does come with one big downside: While copying a dispatcher within the same save is perfectly doable, Blueprinting it for use in other saves or for sharing with others is not. API limitations mean that only the Dispatcher entity is copied, while the configured settings are lost. A second, more minor issue is that you may need to isolate the signals from your inventory with a combinator, or use a different wire, if you wish to use them for other purposes, due RLD's Dispatcher input having negative signals for items it is requesting that might otherwise contaminate them.
```

Comment:
> API limitations mean that only the Dispatcher entity is copied, while the configured settings are lost

This doesn't sound like an API limitation to me. It's should be possible, it's just not easy. Mods like SE do this often because it has many scripted entities.

Reply:
I was taking the mod author's word for it. He's [mentioned the api](https://github.com/dvilker/FactorioRailLogisticsDispatcher/issues/15) [two times](https://mods.factorio.com/mod/RailLogisticsDispatcher/discussion/63da24773837d81618202439) previously. 

Should I just omit any mentions of the API? such as: "Currently, the mod is limited to only copying the dispatcher entity in blueprints, while the configured settings are lost."

I also just noticed him mentioning becoming a father and not having time, which makes me even more hesitant about adding a mention of a GUI in development for RLD.

Reply:
>  But api factorio is not perfect in this regard, and also I am not perfect. And it is also written that other mods can interfere with the process of saving settings and prevent this

I read this as it's hard to implement, but not impossible. And it really is hard to implement. I'm quite an experienced modder and I don't know how to do it.

But yeah, you can mention that the mod doesn't seem to be in active development anymore. This means new features might be limited, and bugs might be present.